doc/lua_api.md

```diff
@@ -6589,6 +6589,9 @@ Bans
   optional reason, this will not prefix with 'Kicked: ' like kick_player.
   If no reason is given, it will default to 'Disconnected.'
     * Returns boolean indicating success (false if player nonexistent)
+* `minetest.transfer_player(name, address, port)`: Disconnect a player and
+  request the client re-connect at a new address and port. The connection
+  to the new server will occur using the same player name and password.
```

Comment:
Nitpick: "will occur using the same player name" - what about the `name` parameter?

Reply:
This is expected to be called by some mod on the server side. So that mod has to identify a player.

As there is name parameter for `minetest.disconnect_player` and `minetest.kick_player`, there is a name parameter for `minetest.transfer_player`.

Reply:
Sorry, I framed that poorly. I understand the API. I just find the wording confusing, since (for obvious reasons) we don't pass a password parameter, yet it says "using the same player name and password". Suggestion: Something along the lines of "[...] will occur using `name` and the client-remembered password" maybe?

Reply:
From my understanding: A password is remembered in some form of hash or something similar. This value is remembered and reused with the player's name to send the AUTH packet to the new server.

Reply:
I think this PR will actually remember the password in plaintext? (Which isn't optimal for security, btw.)

Reply:
You are right. But it looks like it is not only about this PR. There is a field `m_password` in the `Client` class, and I see no place where it is cleared.
May be, it should be a separate issue/PR?

Reply:
I have started working on it. I will create a PR to fix it as soon as possible.
This PR will be forced to wait for that PR to be merged, and I rebase this after it and fix the problem.

Reply:
> I have started working on it. I will create a PR to fix it as soon as possible. This PR will be forced to wait for that PR to be merged, and I rebase this after it and fix the problem.

Won't this PR have to remember the password though to do the SRP auth with the server without requiring the user to re-enter their password? Or is there a way to just remember a hash or something similar? (Which would be ideal.)

Reply:
If there is a way to remember SRP auth without a password, it is an ideal situation. It looks promising.
Otherwise, the client will be forced to reenter the password.

Reply:
See #14196.
Maybe, something more can be done in the main menu Lua scripts.

Reply:
@appgurueu See https://github.com/minetest/minetest/pull/14196#issuecomment-2033989000.